lib/ansible/galaxy/collection/__init__.py

```diff
@@ -1206,7 +1206,8 @@ def _build_files_manifest_walk(b_collection_path, namespace, name, ignore_patter
     def _walk(b_path, b_top_level_dir):
         for b_item in os.listdir(b_path):
             b_abs_path = os.path.join(b_path, b_item)
-            b_rel_base_dir = b'' if b_path == b_top_level_dir else b_path[len(b_top_level_dir) + 1:]
+            common_prefix = os.path.commonprefix([b_top_level_dir, b_path])
+            b_rel_base_dir = b_path[len(common_prefix):].lstrip(b'/')
```

Comment:
[issue (blocking)][conventional comments]: This would require a regression test.

[conventional comments]: https://conventionalcomments.org/

Reply:
Just a quick check, running regression tests mean some extra e2e + unit, but not something to be added to this PR, right?

Reply:
@ccamacho Thanks for checking in!

This PR must contain a new regression test. It is expected that you make one that fails without this patch and succeeds after it's applied.

Normally, it'd be an integration test but I think that for this change, a unit test should be enough. And it'd be simpler.

This doc should help you get started: https://docs.ansible.com/ansible/latest/dev_guide/testing_units.html. Oftentimes, locating existing related tests is useful as you can start by copying or editing some and checking what happens. The style and structure of those can inspire and shape how the new test would look.

Reply:
++ Let me update the test, the nit should be also somewhere there.

Reply:
Marking this thread as resolved.